test/invalid.js

```diff
@@ -1,42 +1,40 @@
-
 // NOTES:
 //  - the errors actually still need to be reviewed to check that they
 //    are fully correct interpretations of the IDLs
 
-var wp = process.env.JSCOV ? require("../lib-cov/webidl2") : require("../lib/webidl2")
-,   expect = require("expect")
-,   pth = require("path")
-,   fs = require("fs")
-;
-describe("Parses all of the invalid IDLs to check that they blow up correctly", function () {
-    var dir = pth.join(__dirname, "invalid/idl")
-    ,   skip = {}
-    ,   idls = fs.readdirSync(dir)
-                  .filter(function (it) { return (/\.w?idl$/).test(it) && !skip[it]; })
-                  .map(function (it) { return pth.join(dir, it); })
-    ,   errors = idls.map(function (it) { return pth.join(__dirname, "invalid", "json", pth.basename(it).replace(/\.w?idl/, ".json")); })
-    ;
+"use strict";
+
+const wp = require("../lib/webidl2");
+const expect = require("expect");
+const pth = require("path");
+const fs = require("fs");
+
+describe("Parses all of the invalid IDLs to check that they blow up correctly", () => {
+  const dir = pth.join(__dirname, "invalid/idl");
+  const skip = {};
```

Comment:
the `skip` should probably be a `new Set()`. I'm not sure `skip` very useful tho, as all the names are going to be unique, right?

Reply:
Probably an array will be sufficient enough as `new Set()` will eventually be written as  `new Set(["testname1", "testname2"])`.

>I'm not sure skip very useful tho, as all the names are going to be unique, right?

I'm sorry but not sure why the uniqueness makes `skip` not useful, can you clarify more?

Reply:
I guess the question is, under what circumstances will there be duplicate file names? (Or have I misunderstood what `skip` doing?) If the answer is "never", then the uniqueness check is not needed.

Reply:
Well, I think `skip` is for temporarily disabling a test which cannot be fixed in the short term, or just a test-to-do-later.

Reply:
I'm ok to keep it as is. If you are happy with it, I'm happy with it :)

Reply:
I think the uniqueness is even an assumption for `skip` as it would accidentally skip multiple tests if names are not unique.